src/main/java/objects/battle/enemy/gimmick/UseGimmick.java

```diff
@@ -0,0 +1,60 @@
+package objects.battle.enemy.gimmick;
+
+import objects.battle.Skill;
+import objects.battle.SkillType;
+
+public class UseGimmick {
```

Comment:
Recommend to rename to GimmickInteractor/GimmickStrategy for clarity

Also recommend to have this as an interface so that you can implement any kind of Gimmick class. 

So like a interface called GimmickStrategy and a concrete class called StatGimmickStrategy or something (also rename previous gimmick class to something more specific like StatGimmickEntity)